go/src/socialapi/models/channel.go

```diff
@@ -176,7 +176,30 @@ func (c *Channel) Create() error {
 	}
 
 	// if we couldnt find the record in the db, then create it
-	return bongo.B.Create(c)
+	return idempotentCreate(c)
+}
+
+func idempotentCreate(c *Channel) error {
+	err := bongo.B.Create(c)
+	if err == nil {
+		return nil
+	}
+
+	if !IsUniqueConstraintError(err) {
+		return err
+	}
+	q := &request.Query{
+		GroupName: c.GroupName,
+		Type:      c.TypeConstant,
+		Name:      c.Name,
+	}
+	ch, er := c.ByName(q)
+	if er != nil {
+		// ignore error for fetch request
+		return err
+	}
+	c = &ch
```

Comment:
The c here is a copy of a pointer value, writing it has effect only within the scope (https://play.golang.org/p/7wch4Wj5Re).

You can either:

```go
*c = ch
```

or change the func sig to `func idempotentCreate(c *Channel) (*Channel, error) {`.